nasa/earthdata-search
# EDSC-3966: Single Harmony service selection box

# Overview

### What is the feature?

A single selection box is shown for available Harmony transformation services rather than a box for each service. This also improves the access method selection UX through simplified language, additional icons, and tooltips.

### What is the Solution?

Users select the Harmony type of access method in step 1 of the access method panel, then the specific Harmony method and its options in step 2.

No changes to the URL routes were necessary, so bookmarked URLs or saved projects with a selected access method should continue to work.

### Example screenshots

Using the collection `C1996881146-POCLOUD`:

| Before |
| ------------- |
| <img width="350" alt="access methods: before" src="https://github.com/nasa/earthdata-search/assets/10775395/2c3e78d7-d623-46f7-830e-121b761a250b">  

| After (Service List) | After (Service Selected) |
| ------------- | ------------- |
| <img width="365" alt="access methods: after, selection box" src="https://github.com/nasa/earthdata-search/assets/10775395/f1673bec-674b-4071-b839-139d5826166e"> | <img width="400" alt="access methods: after" src="https://github.com/nasa/earthdata-search/assets/10775395/7314550c-e948-4d05-8813-602a388b76f8">  |

### What areas of the application does this impact?

Selecting access methods and options before downloading selected data.

# Testing

After running this branch locally:

- Open http://localhost:8080/projects?p=C1996881146-POCLOUD!C1996881146-POCLOUD
- Verify access method details appear as a tooltip over the (?) icons on the right side of each item in Step 1
- Select `Customize with OPeNDAP` and `Download all data` and verify those methods continue to work as before
- Select `Customize with Harmony`
- Click the service selction box in Step 2
- Verify four Harmony methods appear in the selection popup, and you can scroll the popup to view all four
- Verify each of the Harmony methods in the selection popup have icons in their top right corner with tooltips on mouseover that indicate their customization capabilities
- Select a Harmony method from the selection popup
- Verify the service name appears in the selection box
- Verify its name, description, and a link to the Harmony documentation appears below the selection box
- Verify that the appropriate customization options appear below the link to the documentation

# Checklist
